methods.js

```diff
@@ -0,0 +1,385 @@
+   let result = callback(this[i]);
+   if (result) {
+    arrOutPut.push(this[i]);
+   }
+ }
+ return arrOutPut;
+}
+
+const evenNum = function(num){
+  return num % 2 === 0;
+};
+
+console.log(evenNum(7));
+console.log([6,5,3,2,8,9].myFilter(evenNum));
+
+// 10. Use `myEvery` to check if all elements in the array are the same.
+
+Array.prototype.myEvery = function(callback) {
+  for (let i = 0; i < this.length; i++) {
+    if (!callback(this[i])) {
+      return false;
+    }
+  }
+  return true;
+};
+
+const evenNum = function(num){
+  return num % 2 === 0;
+};
+
+console.log(evenNum(4));
+console.log([6,5,3,2,8,9].myEvery(evenNum));
+
+// 11. Use `myReduce` to return the sum of every element in an array.
+
+Array.prototype.myReduce = function(callback, acc) {
+  let startingIdx = 0;
+  if(!acc){
+    acc = this[0];
+    startingIdx++
+  };
+  for (let i = startingIdx; i < this.length; i++) {
+   acc = callback(acc, this[i], i)
+  };
+  return acc;
+};
+
+let arr = [15,3,21,3,8];
+console.log(arr.myReduce((sum, el) => { return sum + el;
+ }));
+
+// 12. Use `myMap` to build a new array that doubles each element. Then chain `myReduce` to find the product of all the elements.
+Array.prototype.myMap2 = function(callback){
+  let arrOutPut = [];
+  for (let i = 0; i < this.length; i++) {
+    arrOutPut.push(callback(this[i]));
+  }
+  return arrOutPut;
+};
+
+Array.prototype.myReduce = function(callback, acc) {
+  let startingIdx = 0;
+  if(!acc){
+    acc = this[0];
+    startingIdx++
+  };
+  for (let i = startingIdx; i < this.length; i++) {
+   acc = callback(acc, this[i], i)
+  };
+  return acc;
+};
+
+let arr = [5,3,2,3];
+let doubleProduct = arr.myMap2((el) => { return el * 2;
+}).myReduce((acc, num) => {
+return acc * num;
+});
+console.log(doubleProduct);
+
+// 13. Use `myReduce` to find the largest number in an array.
+Array.prototype.myReduce = function(callback, acc) {
+  let startingIdx = 0;
+  if(!acc){
+    acc = this[0];
+    startingIdx++
+  };
+  for (let i = startingIdx; i < this.length; i++) {
+   acc = callback(acc, this[i], i)
+  };
+  return acc;
+};
+
+let arr = [15,3,21,3,8];
+
+let largestNum =
+arr.myReduce((acc, el) => {
+if(acc > el){
+  return acc;
+}else if(acc < el){
+  return el;
+};
+});
+
+// 14. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.
+// Use `myFilter` to return a new array of every element of arr that can be evenly divided by divisor.
+
+
+Array.prototype.myFilter = function(callback) {
+let arrOutPut = [];
+ for (let i = 0; i < this.length; i++) {
+   let result = callback(this[i]);
+   if (result) {
+    arrOutPut.push(this[i]);
+   }
+ }
+ return arrOutPut;
+}
+
+let arr = [1,2,3,4,5,6,7,8,9];
+
+const elementDivisibleBy = (divisor, arr) => {
+  let divided = arr.myFilter(el => typeof el === 'number' && !(el % divisor));
+  return divided;
+};
+
+console.log(elementDivisibleBy(3, arr));
+
+
+// 14. Write a `myJoin` function.
+Array.prototype.myJoin = function (callback) {
```

Comment:
it's a little bit confusing to call this a 'callback', right? because it ain't a function, it's a string

Reply:
that being said, this function totally works, so 💯